luigi/contrib/hive.py

```diff
@@ -87,6 +98,19 @@ def run_hive_script(script):
     return run_hive(['-f', script])
 
 
+def _validate_partition(partition):
+    """
+    If partition is set and it's size more than one and it's not ordered
+    than we're unable to restore its path in warehouse definitely
+    """
+    if (
+            partition
+            and len(partition) > 1
+            and not isinstance(partition, collections.OrderedDict)
```

Comment:
how does this handle the fact that python 3.7+ dictionaries guarantee insertion order?

Reply:
Agree.
I should treat that case separately
sth like
`def _is_ordered_dict(dikt):
    import sys
    if sys.version_info >= (3, 7):
        return isinstance(dikt, dict)
    return isinstance(dikt, collections.OrderedDict)


def _validate_partition(partition):
    """
    If partition is set and its size is more than one and not ordered,
    then we're unable to restore its path in the warehouse
    """
    if (
            partition
            and len(partition) > 1
            and _is_ordered_dict(partition)`

Agree?
@dlstadther

Reply:
I think collections.OrderedDict still exists in py3.7 so we'd still want to support that. If we want to allow both, i'd think check for ordereddict first then elif 3.7 and dict. else fail.

Reply:
@dlstadther
I added handling of dicts in py3.7+ and modified a unit-test `test_table_exists_ambiguous_partition` for that
Could you please have a look at that?

Reply:
Do we need to check if order exists when we require one or less elements?

Reply:
@stroykova this code in comments is sketch, have a look at "Files changed" tab, please